spec/Section 7 -- Response.md

```diff
@@ -89,37 +108,25 @@ found at `["hero", "friends"]`, the hero's first friend at
 
 ### Data
 
-The {"data"} entry in the response will be the result of the execution of the
-requested operation. If the operation was a query, this output will be an object
-of the query root operation type; if the operation was a mutation, this output
-will be an object of the mutation root operation type.
+The {"data"} entry in the _execution result_ will be the result of the execution
+of the requested operation. If the operation was a query, this output will be an
+object of the query root operation type; if the operation was a mutation, this
+output will be an object of the mutation root operation type.
 
 The response data is the result of accumulating the resolved result of all
 response positions during execution.
 
-If an error was raised before execution begins, the {"data"} entry should not be
-present in the response.
+If an error was raised before execution begins, the _response_ must be a
+_request error result_. The {"data"} entry must not be present in this map.
 
 If an error was raised during the execution that prevented a valid response, the
 {"data"} entry in the response should be `null`.
 
 ### Errors
 
-The {"errors"} entry in the response is a non-empty list of errors raised during
-the _request_, where each error is a map of data described by the error result
-format below.
-
-If present, the {"errors"} entry in the response must contain at least one
-error. If no errors were raised during the request, the {"errors"} entry must
-not be present in the response.
-
-If the {"data"} entry in the response is not present, the {"errors"} entry must
-be present. It must contain at least one _request error_ indicating why no data
-was able to be returned.
-
-If the {"data"} entry in the response is present (including if it is the value
-{null}), the {"errors"} entry must be present if and only if one or more
-_execution error_ was raised during execution.
+The {"errors"} entry in the _execution result_ or _request error result_ is a
```

Comment:
We could also define a type _Response Result_ that includes both _Execution Result_ and _Request Error Result_ as subtypes, but adding another layer of hierarchy might be less clear?